src/test/kotlin/coverosR3z/authentication/AuthenticationBDD.kt

```diff
@@ -21,29 +18,71 @@ import coverosR3z.domainobjects.LoginStatuses as ls
  */
 class AuthenticationBDD {
 
+    val cua = CurrentUserAccessor() // since we have a method to clear this, we can share it between tests
     /**
      * The key here is that when we do this, it's
      * recording who does what, and only allows proper
      * people to carry out actions.  Like, Alice
      * can only modify Alice's stuff.
      */
+
+    fun `given I am logged in`(pmd: PureMemoryDatabase) {
+        cua.clearCurrentUserTestOnly() // We need to use cua for this test, sharing a pmd with auth persistence
+        // and time recording persistence, in order to avoid recordTime throwing a USER_EMPLOYEE_MISMATCH status
+        val authPersistence = AuthenticationPersistence(pmd)
+        val au = AuthenticationUtilities(authPersistence)
+        au.register(DEFAULT_USER.name, DEFAULT_PASSWORD, 1) // we know DEFAULT_EMPLOYEE has an id=1
+        au.login(DEFAULT_USER.name, DEFAULT_PASSWORD)
+
+        // Perform some quick checks
+        val user = authPersistence.getUser(UserName(DEFAULT_USER.name))
+
+        println("DEBUG")
+        println(cua.get())
+
+        assertEquals(user, cua.get()) // auth persistence and user persistence must agree
+        assertTrue("our user should be registered", au.isUserRegistered(DEFAULT_USER.name)) // registration must succeed
+    }
     @Test
     fun `I can add a time entry`() {
         // given I am logged in
+        val pmd = PureMemoryDatabase()
```

Comment:
In the BDD tests we want to refactor them so it's really easy to read each test and see what is the business intention.

Reply:
I am not sure what _specifically_ this comment is referring to

Reply:
This test has a lot of lines